src/main/scala/io/epiphanous/flinkrunner/util/AvroUtils.scala

```diff
@@ -121,6 +121,29 @@ object AvroUtils extends LazyLogging {
         .foldLeft(instance) { (a, field) =>
           val f = field.name()
           genericRecord.get(f) match {
+            case array: java.util.List[_] =>
```

Comment:
Feels like we should DRY up this block with the similar code starting at L148.

Reply:
Agreed, DRYed up the implementation.